spring-test/src/main/java/org/springframework/test/context/jdbc/SqlScriptsTestExecutionListener.java

```diff
@@ -390,19 +392,10 @@ private Stream<Method> getSqlMethods(Class<?> testClass) {
 		return Arrays.stream(ReflectionUtils.getUniqueDeclaredMethods(testClass, sqlMethodFilter));
 	}
 
-	private void registerClasspathResources(RuntimeHints runtimeHints, String... locations) {
-		Arrays.stream(locations)
-				.filter(location -> location.startsWith(CLASSPATH_URL_PREFIX))
-				.map(this::cleanClasspathResource)
-				.forEach(runtimeHints.resources()::registerPattern);
-	}
-
-	private String cleanClasspathResource(String location) {
-		location = location.substring(CLASSPATH_URL_PREFIX.length());
-		if (!location.startsWith(SLASH)) {
-			location = SLASH + location;
-		}
-		return location;
+	private void registerClasspathResources(RuntimeHints runtimeHints, ClassLoader classLoader, String... locations) {
+		DefaultResourceLoader resourceLoader = new DefaultResourceLoader(classLoader);
+		RuntimeHintsUtils.registerResources(runtimeHints,
+			TestContextResourceUtils.convertToResources(resourceLoader, locations));
```

Comment:
For the above to be useful in a more broader context, something that lets you convert locations to `Resources` outside of `spring-test` would be needed.

Reply:
If we leave out `registerResources()` (plural), this becomes a moot point.

I'm not sure if such a utility would be used very often outside `spring-test`. Nobody has requested it so far, so maybe it's better to leave it as-is for the time being.